src/uwtools/rocoto.py

```diff
@@ -106,25 +106,24 @@ def realize_rocoto_xml(
     :return: Did the input and output files conform to theirr schemas?
     """
 
-    # Validate the YAML.
-    if uwtools.config.validator.validate_yaml(
-        config_file=config_file, schema_file=_rocoto_schema_yaml()
-    ):
-        # Render the template to a temporary file.
-        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
-            _write_rocoto_xml(
-                config_file=config_file,
-                rendered_output=temp_file.name,
-            )
-            # Validate the XML.
-            if validate_rocoto_xml(input_xml=temp_file.name):
-                # If no issues were detected, save temp file and report success.
-                shutil.move(temp_file.name, rendered_output)
-                return True
+    if not validate_yaml(config_file=config_file, schema_file=_rocoto_schema_yaml()):
+        log.error("YAML validation errors identified in %s", config_file)
+        return False
+
+    temp_file = tempfile.NamedTemporaryFile(delete=False)  # pylint: disable=consider-using-with
+    _write_rocoto_xml(config_file=config_file, rendered_output=temp_file.name)
+
+    if not validate_rocoto_xml(input_xml=temp_file.name):
         log.error("Rocoto validation errors identified in %s", temp_file.name)
         return False
-    log.error("YAML validation errors identified in %s", config_file)
-    return False
+
+    if rendered_output is None:
+        with open(temp_file.name, "r", encoding="utf-8") as f:
+            print(f.read())
+        Path(temp_file.name).unlink()
+    else:
+        Path(temp_file.name).rename(rendered_output)
```

Comment:
Why not use "writeable" here to stay consistent with other "dump" methods (untested):

```
with open(temp_file.name, "r", encoding="utf-8") as f:
    content = f.read()
    with writable(rendered_output):
        print(content, file=f)
```

I think that may also help with the context manager and will clean up once this is done.

Reply:
Hmm - from the [discussion](https://github.com/ufs-community/workflow-tools/pull/324/#discussion_r1370785648) above with @maddenp-noaa that led to this, it made sense at the time to leave the writeable and context manager out for simplicity, although its a fair point to maintain consistency with other functions. It should work fine either way

Reply:
Expanding on what I mentioned in the comment I made a few minutes ago, I think we cannot use the context manager because, if XML validation fails, the file needs to be left on disk for the user to inspect, and the context manager will always remove it. We could perhaps rename it to prevent the context manager from deleting it (if that works), but the whole point of using these methods / classes from Python's `tempfile` package is to ensure that a new temporary file does not conflict with an existing file, so it's not safe to arbitrarily rename files. We should only use filenames that the user supplies (if something gets clobbered, that's on them), or techniques that guarantee that we won't clobber something (like the `tempfile` tools do).

I think we could do, say,
```
with open(temp_file, "r", encoding="utf-8") as f_in:
    with writable(rendered_output) as f_out:
        print(f_in.read(), file=f_out)
Path(temp_file).unlink()
```
but we'd be opening, reading, and writing to create a file identical to `temp_file`, which we could just rename. I do like the simplicity and clarity of this approach, and maybe doing some tiny unnecessary file IO is worth it for that. Thoughts?

Reply:
My preference is to keep it simpler, given the choice

Reply:
Yes, I think I prefer the simpler version, too. Trying to avoid the extra file IO is premature optimization. So, I think the code block I suggested above is worth a try.

Reply:
Sure thing. It's nice to have the discussion of pros/cons of alternatives.